<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FPGA 调试上位机</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="leftPanelWidget" native="true">
       <property name="minimumSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>250</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QGroupBox" name="connectionSettingsBox">
          <property name="title">
           <string>通信设置</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>模式:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="communicationModeComboBox">
               <item>
                <property name="text">
                 <string>串口</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>TCP 客户端</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>UDP</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>TCP 服务器</string>
                </property>
               </item>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QStackedWidget" name="settingsStackedWidget">
             <property name="currentIndex">
              <number>2</number>
             </property>
             <widget class="QWidget" name="serialSettingsPage">
              <layout class="QFormLayout" name="formLayout">
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>串口号</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QComboBox" name="portComboBox"/>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>波特率</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QComboBox" name="baudRateComboBox"/>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>校验位</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QComboBox" name="parityComboBox"/>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>数据位</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QComboBox" name="dataBitsComboBox"/>
               </item>
               <item row="4" column="0">
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>停止位</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="1">
                <widget class="QComboBox" name="stopBitsComboBox"/>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tcpSettingsPage">
              <layout class="QFormLayout" name="formLayout_2">
               <item row="0" column="0">
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>服务器 IP:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="tcpHostLineEdit">
                 <property name="text">
                  <string>192.168.1.10</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_9">
                 <property name="text">
                  <string>端口:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSpinBox" name="tcpPortSpinBox">
                 <property name="maximum">
                  <number>65535</number>
                 </property>
                 <property name="value">
                  <number>8080</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="udpSettingsPage">
              <layout class="QFormLayout" name="formLayout_3">
               <item row="0" column="0">
                <widget class="QLabel" name="label_10">
                 <property name="text">
                  <string>本地端口:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="udpBindPortSpinBox">
                 <property name="maximum">
                  <number>65535</number>
                 </property>
                 <property name="value">
                  <number>8888</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_11">
                 <property name="text">
                  <string>目标 IP:</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="udpTargetHostLineEdit">
                 <property name="text">
                  <string>192.168.1.10</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_12">
                 <property name="text">
                  <string>目标端口:</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QSpinBox" name="udpTargetPortSpinBox">
                 <property name="maximum">
                  <number>65535</number>
                 </property>
                 <property name="value">
                  <number>8080</number>
                 </property>
                </widget>
               </item>
               <item row="3" column="0" colspan="2">
                <widget class="QCheckBox" name="useWinSockCheckBox">
                 <property name="text">
                  <string>使用原生Windows Socket</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="tcpServerSettingsPage">
              <layout class="QFormLayout" name="formLayout_4">
               <item row="0" column="0">
                <widget class="QLabel" name="label_13">
                 <property name="text">
                  <string>监听端口:</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="tcpListenPortSpinBox">
                 <property name="maximum">
                  <number>65535</number>
                 </property>
                 <property name="value">
                  <number>8888</number>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="connectButton">
             <property name="text">
              <string>打开</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="hostComputerFunctionsBox">
          <property name="title">
           <string>上位机功能</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QPushButton" name="sendTextAsFileButton">
             <property name="text">
              <string>发送文本内容</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QFormLayout" name="formLayout_5">
             <item row="0" column="0">
              <widget class="QLabel" name="label_packetSize">
               <property name="text">
                <string>每包字节:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBox" name="packetSizeSpinBox">
               <property name="maximum">
                <number>65535</number>
               </property>
               <property name="value">
                <number>1400</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_delay">
               <property name="text">
                <string>分包延时(ms):</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="delaySpinBox">
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="value">
                <number>10</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="sendBigFileButton">
             <property name="text">
              <string>发送文件</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>已连接客户端:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QListWidget" name="clientListWidget"/>
           </item>
           <item>
            <widget class="QPushButton" name="disconnectClientButton">
             <property name="text">
              <string>断开选定客户端</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="receiveSettingsBox">
          <property name="title">
           <string>数据显示设置</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_receiveMode">
             <item>
              <widget class="QLabel" name="label_receiveMode">
               <property name="text">
                <string>接收模式:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="autoDisplayRadio">
               <property name="text">
                <string>自动</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="textDisplayRadio">
               <property name="text">
                <string>文本</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="imageDisplayRadio">
               <property name="text">
                <string>图像</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="videoDisplayRadio">
               <property name="text">
                <string>视频</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QRadioButton" name="asciiDisplayRadio">
               <property name="text">
                <string>ASCII</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="hexDisplayRadio">
               <property name="text">
                <string>HEX</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="decimalDisplayRadio">
               <property name="text">
                <string>十进制</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QCheckBox" name="autoWrapCheckBox">
             <property name="text">
              <string>日志区自动换行</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="clearReceiveButton">
             <property name="text">
              <string>清空日志和计数</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="sendSettingsBox">
          <property name="title">
           <string>文本发送设置</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QRadioButton" name="asciiSendRadio">
               <property name="text">
                <string>ASCII</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="hexSendRadio">
               <property name="text">
                <string>HEX</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QCheckBox" name="cyclicSendCheckBox">
             <property name="text">
              <string>循环周期</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QSpinBox" name="sendIntervalSpinBox">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="maximum">
                <number>1000000</number>
               </property>
               <property name="value">
                <number>1000</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>ms</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="rightPanelWidget" native="true">
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QGroupBox" name="imageDisplayBox">
          <property name="title">
           <string>图像/视频显示</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_9">
           <item>
            <widget class="QStackedWidget" name="displayStackedWidget">
             <property name="currentIndex">
              <number>1</number>
             </property>
             <widget class="QWidget" name="videoPage">
              <layout class="QVBoxLayout" name="verticalLayout_10">
               <property name="spacing">
                <number>0</number>
               </property>
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QVideoWidget" name="videoDisplayWidget" native="true">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                   <horstretch>0</horstretch>
                   <verstretch>1</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">background-color: rgb(0, 0, 0);</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="imagePage">
              <layout class="QGridLayout" name="gridLayout">
               <item row="0" column="0">
                <widget class="QLabel" name="imageDisplayLabel">
                 <property name="styleSheet">
                  <string notr="true">background-color: rgb(0, 0, 0);</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="resolutionLabel">
                 <property name="styleSheet">
                  <string notr="true">background-color: rgba(0, 0, 0, 150);
color: white;
padding: 2px;</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTop|Qt::AlignTrailing</set>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QPushButton" name="playPauseButton">
               <property name="text">
                <string>播放</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="progressSlider">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="clearDisplayButton">
               <property name="text">
                <string>清除显示</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QTabWidget" name="tabWidget">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="receiveTab">
           <attribute name="title">
            <string>接收日志</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QTextEdit" name="receiveDataDisplayEdit">
              <property name="font">
               <font>
                <family>Consolas</family>
               </font>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="sendTab">
           <attribute name="title">
            <string>发送日志</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QTextEdit" name="sentDataDisplayEdit">
              <property name="font">
               <font>
                <family>Consolas</family>
               </font>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="sendDataEdit">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>100</height>
           </size>
          </property>
          <property name="placeholderText">
           <string>输入要发送的文本数据...</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="sendButton">
            <property name="text">
             <string>发送</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QVideoWidget</class>
   <extends>QWidget</extends>
   <header>QVideoWidget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>cyclicSendCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>sendIntervalSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>596</y>
    </hint>
    <hint type="destinationlabel">
     <x>93</x>
     <y>625</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>